### ISSUE TITLE
InputControl misses to set a default color for the placeholder text

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#69331]
### Description

Discovered while working on https://github.com/WordPress/gutenberg/issues/60275

Not sure what the plans are for the `InputControl` and `TextareaControl` components, especially regarding the CSS-in-JS part. I seem to recall there was an ongoing discussion about not using CSS-in-JS for the future so this may need to wait for a decision.

Anyways:

`TextareaControl` does set a default color for the placeholder text:

https://github.com/WordPress/gutenberg/blob/7a4b5aa5d8a63d97b084e1339e6ce544dfc944bb/packages/components/src/textarea-control/styles/textarea-control-styles.ts#L59-L71

`InputControl` doesn't, so it uses the browser default:

https://github.com/WordPress/gutenberg/blob/7a4b5aa5d8a63d97b084e1339e6ce544dfc944bb/packages/components/src/input-control/styles/input-control-styles.tsx

It's important to remind that the placeholder text should have a sufficient color contrast. That's the reason why Core and the Editor provide CSS to set the color by targeting the prefixed CSS selectors for -webkit, -moz, and -ms.

I would say the two components should behave consistently. They either should provide a default color or should not and delegate the styling to the consumers.

Note: according to https://github.com/WordPress/gutenberg/issues/69311 the `opacity` is no longer needed for Firefox.

Cc @WordPress/gutenberg-components 

### Step-by-step reproduction instructions

- Go to the InputControl in the Storybook: https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-inputcontrol--docs
- Compare the placeholder text color in Chrome and Firefox.
- Observe the color is slightly different. Use a color measurement tool if necessary e.g. Digital Color Meter on macOS, as the color difference is hard to perceive for the eye.
- Go to the TextareaControl in the Storybook: https://wordpress.github.io/gutenberg/?path=/docs/components-textareacontrol--docs
- Compare the placeholder text color in Chrome and Firefox.
- Observe it's the same color, because it's provided by the TextareaControl component.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by SainathPoojary** [comment]
Hey @afercia,

Thanks for bringing this up! I was able to reproduce the issue in Storybook and will be raising a PR shortly.

Placeholder color
Firefox: `#858585`
Chrome: `#757575`

**Comment by ciampo** [comment]
Aligning implementations seems reasonable, regardless of CSS-in-JS or not

**Comment by im3dabasia** [comment]
Thank you @afercia 

This is possibly related to https://github.com/WordPress/gutenberg/issues/64666, and this PR aims to address the issue described there: https://github.com/WordPress/gutenberg/pull/68807.

**Comment by afercia** [comment]
@im3dabasia oh I see, there are now two PRs that slightly overlap. Any objection to merging https://github.com/WordPress/gutenberg/pull/69334 ? I have an idea for further improvements that could be implemented in the context of https://github.com/WordPress/gutenberg/issues/64666 / https://github.com/WordPress/gutenberg/pull/68807

**Comment by im3dabasia** [comment]
@afercia Thank you! Yes, we can merge [#69334](https://github.com/WordPress/gutenberg/pull/69334) first, and then merge [#68807](https://github.com/WordPress/gutenberg/pull/68807) after addressing the improvements you suggest.

Looping in @t-hamano.
